core/templates/pages/exploration-editor-page/translation-tab/audio-translation-bar/audio-translation-bar.directive.ts

```diff
@@ -573,14 +567,11 @@ angular.module('oppia').directive('audioTranslationBar', [
                 () => $scope.openTranslationTabBusyModal()
               )
             );
-            $scope.track = {
-              progress: function(progressPercentage) {
-                if (angular.isDefined(progressPercentage)) {
-                  AudioPlayerService.setProgress(progressPercentage / 100);
-                }
-                return AudioPlayerService.getProgress() * 100;
-              }
-            };
+            ctrl.directiveSubscriptions.add(
+              AudioPlayerService.viewUpdate.subscribe(() => {
+                $scope.$applyAsync();
```

Comment:
over here we only want to update the audio duration, right? if yes then can we just update that specific value in this controller?

Reply:
We also need to tell the controller to update the view. I have tried this, the slider and the timer don't update even though the value in the controller is different :(